xbmc/FileItem.cpp

```diff
@@ -124,6 +124,10 @@ CFileItem::CFileItem(const CEpgInfoTagPtr& tag)
   Initialize();
 
   m_bIsFolder = false;
+  if (tag->HasPVRChannel())
+  {
+    m_pvrChannelInfoTag = tag->ChannelTag();
```

Comment:
I'm afraid that this may cause serious strange side affects, because usually a file item contains at most one of the pvr types! Using code assumes that there are no "multitype" file items.